content/3_Silver/2P.mdx

```diff
@@ -31,6 +31,139 @@ frequency: 2
 	</Resource>
 </Resources>
 
+## Two Pointers
+
+The two pointers method iterates two pointers across an array, to track the start and end
+of an interval, or two values in a sorted array that we are currently checking. Both pointers
+are monotonic; meaning each pointer starts at one end of the array and only moves in one
+direction.
+
+<FocusProblem problem="book" />
+
+## Solution - Books
+
+We want to find the longest continuous segment of books that can be read within $t$ minutes. 
+
+To accomplish this, we can define $\texttt{left}$ and $\texttt{right}$ as the start and end of the segment respectively.
+
+Since being able to read segment $\texttt{left} \dots \texttt{right}$ implies we can read segments $\texttt{left} \dots \texttt{right}-1$, we can use two-pointer (or, alternatively, [binary search](/silver/binary-search)).
+
+For every $\texttt{left}$ index, we need to increase our $\texttt{right}$ pointer until the total time for that segment exceeds $t$.
+
+We will keep track of the total time with $\texttt{cur}$.
+
+Once $\texttt{cur}$ exceeds $t$, we remove the current $\texttt{right}$-indexed book and decrement $\texttt{right}$ by 1.
+
+We will keep track of the max segment size with $\texttt{ans}$, and take the maximum between $\texttt{ans}$ and $\texttt{right} - \texttt{left}$.
```

Comment:
also kind of self-explanatory that we are trying to find max.

Reply:
maybe it's better to reiterate pointers move in one direction and the right pointer simply keeps moving to the right instead of starting over?